### PR TITLE
Add Provider and Configuration package types

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Breaks out adding the `Provider` and `Configuration` package types from https://github.com/crossplane/crossplane/pull/1675 to unblock related work. Commends from #1675 are addressed in review below.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

`make reviewable`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/pkg/v1alpha1/revision_types.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	// Crossplane is a semantic version for supported Crossplane version for the
+	// package.
+	Crossplane string `json:"crossplane,omitempty"`
+
+	// DependsOn is the list of packages and CRDs that this package depends on.
+	DependsOn []Dependency `json:"dependsOn,omitempty"`
+
+	// References to objects owned by PackageRevision.
+	ObjectRefs []runtimev1alpha1.TypedReference `json:"objectRefs,omitempty"`
```
Comment (hasheddan) [inline apis/pkg/v1alpha1/revision_types.go:75]: Resolves https://github.com/crossplane/crossplane/pull/1675#discussion_r487451802

File: apis/pkg/v1alpha1/interfaces.go [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+}
+
+// GetRevisions of this ConfigurationRevisionList.
+func (p *ConfigurationRevisionList) GetRevisions() []PackageRevision {
+	prs := make([]PackageRevision, len(p.Items))
+	for i, r := range p.Items {
+		prs[i] = &r
+	}
+	return prs
+}
```
Comment (hasheddan) [inline apis/pkg/v1alpha1/interfaces.go:421]: Resolves https://github.com/crossplane/crossplane/pull/1675#discussion_r487452206

File: apis/pkg/v1alpha1/revision_types.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+	// PermissionRequests are additional permissions that should be added to a
+	// packaged controller's service account.
+	PermissionRequests []rbac.PolicyRule `json:"permissionRequests,omitempty"`
```
Comment (hasheddan) [inline apis/pkg/v1alpha1/revision_types.go:79]: Resolves https://github.com/crossplane/crossplane/pull/1675#discussion_r487452274

File: apis/pkg/v1alpha1/interfaces.go [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+
+// GetRevisions of this ConfigurationRevisionList.
+func (p *ConfigurationRevisionList) GetRevisions() []PackageRevision {
+	prs := make([]PackageRevision, len(p.Items))
+	for i, r := range p.Items {
+		r := r // Pin range variable so we can take its address.
+		prs[i] = &r
+	}
+	return prs
+}
```
Comment (hasheddan) [inline apis/pkg/v1alpha1/interfaces.go:423]: Resolves https://github.com/crossplane/crossplane/pull/1675#discussion_r487452206

**Review by negz — APPROVED** [review]

**Pull request merged**